tests/test_markers.py

```diff
@@ -303,6 +303,46 @@ def test_extra_with_no_extra_in_environment(self):
                 {"extra": "security"},
                 True,
             ),
+            (
+                "extra == 'security'",
+                {"extra": None},
+                False,
+            ),
+            (
+                "extra == '3.7.0'",
+                {"extra": None},
+                False,
+            ),
+            (
+                "extra == '3.7.0'",
+                {"extra": "foobar"},
+                False,
+            ),
+            (
+                "extra == '3.7.0'",
+                {"extra": "3.7.0"},
+                True,
+            ),
+            (
+                "extra == '3.7.0'",
+                {"extra": "3.7"},
+                True,
+            ),
+            (
+                "extra == '3.7.0'",
+                {"extra": "3.7.0.0"},
+                True,
+            ),
+            (
+                "extra == '3.7.0'",
+                {"extra": "3.7.0+youaregreat"},
+                True,
```

Comment:
I agree, this PR follows the PEP-508, specifically:

```
Comparisons in marker expressions are typed by the comparison operator. The
<marker_op> operators that are not in <version_cmp> perform the same as they
do for strings in Python. The <version_cmp> operators use the PEP-440 version
comparison rules when those are defined (that is when both sides have a valid
version specifier). 
```

but again, I don't think the PEP-508 wanted to allow "non-version" markers (that is `os_name`, `sys_platform`,  `platform_machine`, `platform_python_implementation`, `platform_system`, `implementation_name` and IMHO `extra`) to behave like version numbers.
cc @rbtcollins